src/Workspaces/Core/Portable/Serialization/SerializerService_Reference.cs

```diff
@@ -132,6 +153,9 @@ protected virtual void WriteAnalyzerReferenceTo(AnalyzerReference reference, Obj
                 //    ShadowCopyAnalyzerAssemblyLoader.  This loader will *itself* then use the MVID of the file
                 //    reference at the requested path to shadow copy to a new location specific to that mvid, ensuring
                 //    that its data can be cleanly loaded in isolation from any prior version.
+                //
+                // During testing, we do write it over for validation purposes.
+                writer.WriteGuid(forTesting ? TryGetAnalyzerFileReferenceMvid(file) : Guid.Empty);
```

Comment:
>WriteGuid [](http://example.com/codeflow?start=23&length=9)

nit: doubt we care about amount written, but could write a bool in the normal, non-test case instead of a guid

Reply:
sure.

Reply:
actually, that makes things harder.  now we need the reading code to be exposed as well.  i'd prefer just this :)